zun/container/k8s/driver.py

```diff
@@ -882,3 +908,23 @@ def create_capsule(self, context, capsule, **kwargs):
 
     def delete_capsule(self, context, capsule, **kwargs):
         raise NotImplementedError()
+
+    def get_secrets_for_image(self, image, context):
+        image_parts = docker_image.Reference.parse(image)
+        domain, _ = image_parts.split_hostname()
+        secrets = []
+        for registry in objects.Registry.list(context):
+            if registry.domain == domain:
+                # NOTE this assumes (domain, username) is unique per project
+                name = f"{domain}-{registry.username}"
```

Comment:
Will this always be a valid k8s name? What about just using the registry UUID assigned by Zun?